requirements-optional.txt

```diff
@@ -0,0 +1,40 @@
+# Optional dependencies of Terra that can (mostly) reliably be installed with
+# `pip`.  This file is still called `requirements-optionals.txt` just to match
+# standard pip conventions, even though none of these are required.
+#
+# If updating this, you probably want to update `qiskit.utils.optionals` and
+# maybe `setup.py` too.
+
+# Test-runner enhancements.
+fixtures
```

Comment:
Any downside to installing `fixtures` and `testtools` in `requirements-dev.txt`? Presumably those are helpful when locally running tests.

Reply:
There historically was a reason, particularly for `testtools`, but I don't remember what it was now.  Matthew (@mtreinish) will know.

Reply:
There were two historical reasons for it. First there was a time when there was an incompatibility with some skip decorators between testtools and unittest on newer python versions, but I believe we fixed that in the testtools 2.5.0 release. The second and larger reason was for people he wanted to just run with unittest. The only thing we use testtools for is to wrap unittest and leverage testtools' result stream feature. That lets us attach captured stdout, stderr, and python logging and have that associated with an individual test. The issue was historically there are people who don't like running with stestr and prefer using `make test` to use the stdlib python unittest runner. If we used testtools' stdout, stderr, and logging capture fixtures then that would hide any stdout from tests for people using the stdlib runner. But that should have also been sorted long ago by adding an env variable to enable to the capture.

TBH, this was mostly the result of a historical compromise from when we first started using stestr for qiskit back in 2018/2019. Back then some others felt very strongly that people should be able to run the tests without installing an extra test runner package (neglecting that you needed a bunch of different libraries to even install qiskit, let alone run the tests). So we made started using stestr by default for the improved throughput it provided we made sure there was a path for those who didn't want to abandon the stdlib runner.

That being said 4-5 yrs later I'm not sure this really matters anymore. I think in a follow up we should just move it into the requirements list and see what happens. I would expect anyone still using `make test` (despite the warnings it prints about using `tox` instead) should still have the same experience.